### DISCUSSION TITLE
Form inputs do not reflect Field Array

### DISCUSSION THREAD

**Discussion by jostal** [react-hook-form/react-hook-form#8228] [Q&A] [answered]
I am creating a form which has a dynamic table in it based on data fetched from a db.
In this table, there is an option to set a row as primary. I only want one row to be primary at a time.
Currently, I have written it so the Field Array works as intended - when one checkbox is selected, the other rows are all set to false for isPrimary.
However, despite the field array having the correct info, the checkboxes remain checked so they are not reflecting the data in field array.
Am I using useFieldArray wrong? I thought the field array was supposed to reflect the inputs.
Is there another way I should be doing this?
Suggestions on how to just uncheck the checkboxes without using field array are welcome too.

Here is a code sandbox close to my current setup:
https://codesandbox.io/s/only-one-checkbox-41fl6y

Thanks in advance

**Comment by Moshyfawn** [comment]
`fieldArray` is immutable. You can't do
```js
fieldArray.fields[i].isPrimary = true
```

Check out [RHF docs](https://react-hook-form.com/v6/api#useFieldArray) V6 for form methods to update form's values

**Reply by jostal** [reply]
How can I change the field though? All I'm seeing there is how to add or remove the whole thing which isn't what I want. I'm looking to just change a property of the field

**Reply by Moshyfawn** [reply]
There're a bunch of additional field array methods in V7, that can be convenient in your use-case. With V6, check out my comment below

**Comment by Moshyfawn** [comment, marked as answer]
With RHF V6, I'd suggest to explore registering a `hidden` input as "primaryRow" and use `setValue` `onChange` of checkboxes to set the `primaryRow` field value to `field.id`.
```js
onChange={() => setValue("primaryRow", field.id)}
```

You can `watch` the `primaryRow` value and set the checkboxes value like
```js
checked={primaryRow === field.id}
```

**Reply by jostal** [reply]
I see, this worked wonderfully. Thank you very much

**Reply by Moshyfawn** [reply]
No problem! ;)
